core/templates/components/state-editor/state-editor-properties-services/state-editor.service.spec.ts

```diff
@@ -145,5 +231,323 @@ describe('Editor state service', () => {
+    ecs.setInteractionDefaultOutcome(newDefaultOutcome);
+    expect(ecs.interaction.defaultOutcome).toEqual(newDefaultOutcome);
+  });
+
+  it('should set interaction customization args', () => {
+    let newCustomizationArgs = {
+      rows: {
+        value: 2,
+      },
+      placeholder: {
+        value: suof.createDefault('2', ''),
+      },
+    };
+    ecs.setInteraction(mockInteraction);
+    expect(ecs.interaction.customizationArgs).toEqual({
+      placeholder: {
+        value: suof.createDefault('1', 'cid'),
+      },
+      rows: {
+        value: 1
+      }
+    });
+    ecs.setInteractionCustomizationArgs(newCustomizationArgs);
+    expect(ecs.interaction.customizationArgs).toEqual(newCustomizationArgs);
+  });
+
+  it('should set interaction solution', () => {
+    let newSolution = sof.createFromBackendDict({
+      answer_is_exclusive: true,
+      correct_answer: 'test_answer_new',
+      explanation: {
+        content_id: '2',
+        html: 'test_explanation1_new',
+      },
+    });
+    ecs.setInteraction(mockInteraction);
+    expect(ecs.interaction.solution).toEqual(sof.createFromBackendDict({
+      answer_is_exclusive: true,
+      correct_answer: 'test_answer',
+      explanation: {
+        content_id: '2',
+        html: 'test_explanation1',
+      },
+    }));
+    ecs.setInteractionSolution(newSolution);
+    expect(ecs.interaction.solution).toEqual(newSolution);
+  });
+
+  it('should set interaction hints', () => {
+    let newHints = [hof.createFromBackendDict({
+      hint_content: {
+        content_id: '',
+        html: 'This is a hint'
+      }
+    })];
+    ecs.setInteraction(mockInteraction);
+    expect(ecs.interaction.hints).toEqual([]);
+    ecs.setInteractionHints(newHints);
+    expect(ecs.interaction.hints).toEqual(newHints);
+  });
+
+  it('should set in question mode', () => {
+    expect(ecs.isInQuestionMode()).toBe(null);
+    ecs.setInQuestionMode(true);
+    expect(ecs.isInQuestionMode()).toBe(true);
+    ecs.setInQuestionMode(false);
+    expect(ecs.isInQuestionMode()).toBe(false);
+  });
+
+  it('should set correctness feedback enabled', () => {
+    expect(ecs.getCorrectnessFeedbackEnabled()).toBe(null);
+    ecs.setCorrectnessFeedbackEnabled(true);
+    expect(ecs.getCorrectnessFeedbackEnabled()).toBe(true);
+    ecs.setCorrectnessFeedbackEnabled(false);
+    expect(ecs.getCorrectnessFeedbackEnabled()).toBe(false);
+  });
+
+  it('should set inapplicable skill misconception ids', () => {
+    expect(ecs.getInapplicableSkillMisconceptionIds()).toEqual([]);
+    ecs.setInapplicableSkillMisconceptionIds(['id1', 'id2']);
+    expect(ecs.getInapplicableSkillMisconceptionIds()).toEqual(['id1', 'id2']);
+  });
+
+  it('should check if current solution is valid', () => {
+    ecs.activeStateName = 'Hola';
+    expect(ecs.isCurrentSolutionValid()).toBe(undefined);
+    solutionValidityService.init(['Hola']);
+    expect(ecs.isCurrentSolutionValid()).toBe(true);
```

Comment:
As a reader, it is not clear why is happening in this test. Can you please add comments?

Reply:
Added a comment